File: src/logan/pickpocket/commands/TargetCommand.java
package logan.pickpocket.commands;

import logan.config.MessageConfiguration;
import logan.pickpocket.main.PickpocketPlugin;
import logan.pickpocket.main.Profiles;
import logan.pickpocket.profile.Profile;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.util.List;

public class TargetCommand implements PickpocketCommand {

    @Override
    public <T> void execute(Player player, List<Profile> profiles, T... args) {
        Player victim = Bukkit.getPlayer(args[0].toString());
        if (victim == null) {
            player.sendMessage(PickpocketPlugin.getMessageConfiguration().getMessage(MessageConfiguration.PLAYER_NOT_FOUND_KEY));
            return;
        }
        // Check that the victim can be seen and is at most 5 blocks away from the player
        if (player.getLocation().distance(victim.getLocation()) > 4) {
            player.sendMessage(PickpocketPlugin.getMessageConfiguration().getMessage(MessageConfiguration.PLAYER_NOT_ACCESSIBLE_KEY));
            return;
        }
        Profile playerProfile = Profiles.get(player);
        Profile victimProfile = Profiles.get(victim);
        // Make sure both players have pick-pocketing enabled.
        if (!playerProfile.isParticipating()) {
            player.sendMessage(PickpocketPlugin.getMessageConfiguration().getMessage(MessageConfiguration.PICKPOCKET_DISABLED_KEY));
            return;
        }
        if (!victimProfile.isParticipating()) {
            player.sendMessage(PickpocketPlugin.getMessageConfiguration().getMessage(MessageConfiguration.PICKPOCKET_DISABLED_OTHER_KEY));
            return;
        }
        playerProfile.performPickpocket(victim);
    }
}

File: src/logan/pickpocket/commands/TargetCommand.kt
package logan.pickpocket.commands;

import logan.api.command.BasicCommand
import logan.api.command.SenderTarget
import logan.pickpocket.config.MessageConfiguration
import logan.pickpocket.main.Profiles
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class TargetCommand : BasicCommand<Player>(
    name = "target",
    argRange = 1..1,
    listOf(String::class),
    parentCommand = "pickpocket",
    target = SenderTarget.PLAYER,
    permissionNode = "pickpocket.target",
    usage = """
        Usage:
        /pickpocket target <name> - Pick-pocket player near you.
    """.trimIndent(),
) {

    override fun run(sender: Player, args: Array<out String>, data: Any?): Boolean {
        val victim = Bukkit.getPlayer(args[0]) ?: run {
            sender.sendMessage(MessageConfiguration.getPlayerNotFoundMessage())
            return true;
        }

        // Check that the victim can be seen and is at most 4 blocks away from the player
        if (sender.location.distance(victim.location) > 4) {
            sender.sendMessage(MessageConfiguration.getPlayerNotAccessibleMessage())
            return true
        }

        val senderProfile = Profiles.get(sender)
        val victimProfile = Profiles.get(victim)

        // Make sure both players have pick-pocketing enabled.
        if (!senderProfile.isParticipating) {
            sender.sendMessage(MessageConfiguration.getPickpocketDisabledMessage())
            return true
        }

        if (!victimProfile.isParticipating) {
            sender.sendMessage(MessageConfiguration.getPickpocketDisabledOtherMessage())
            return true
        }

        senderProfile.doPickpocket(victimProfile)
        return true
    }
}